### ISSUE TITLE
Should -Invoke test from 5.1.1 fails on 5.2.2

### ISSUE THREAD

**Issue by efie45** [pester/Pester#1991]
## General summary of the issue

Using Pester 5.1.1 I had something like this:

```powershell
It "calls Write-Progress" {
            $tracker = [myCustomClass]::new(@(1))
            Should -Invoke Write-Progress -Exactly 1
}
```

The constructor from [myCustomClass] would call a function that called another function which called Write-Progress. In 5.2.2 the above test fails where it would pass previously. What is going on here and how do I fix it?

## Describe your environment
Pester version     : 5.2.2 /Users/____/.local/share/powershell/Modules/Pester/5.2.2/Pester.psm1                                                                                       PowerShell version : 7.1.3                                                                   OS version         : Unix 11.0.0

**Comment by nohwnd** [comment]
> What do you specify for Mock and Should -Invoke when you want to see how many times something was called from _**any**_ module?

There is no such option, but I was proposing to do it by *, if there is a need to do that. I don't think we will be able to get it to 5.3, there is already a lot planned for that milestone. But if you or @fflaten want to implement it go ahead, I will gladly merge it early.

**Comment by fflaten** [comment]
> There is no such option, but I was proposing to do it by *, if there is a need to do that.

Seems logical as an option in `Should -Invoke` to quickly assert globally. Wildcard search in the mock call history should be doable.

Or did you also consider it in Mock as "publish in all modules"/fallback behavior (have I seen this in another issue or PR comment?)?

**Comment by nohwnd** [comment]
I did not. Seems like overkill that would be rarely used, and would be very costly to use as we would have to define hooks in all modules that are currently loaded.

**Comment by fflaten** [comment]
Exactly, just checking because I believe we mentioned this as "fallback" to look into when making -ModuleName more strict in 5.2.

I might take a look at the Should Invoke option unless @efie45 wants to give it a shot. 🙂

**Comment by nohwnd** [comment]
> Wildcard search in the mock call history should be doable.

Yeah that should work just fine, you just need to avoid filtering down to the mocks that are targetting the given module. I don't remember where exactly we do that, but if you stop in a mock filter when calling Should -Invoke you will be able to find it easily in the call stack I think.

**Comment by efie45** [comment]
I'd love to give it a shot. I've never contributed before though so it may take me a little longer

**Comment by nohwnd** [comment]
@efie45 Assigning it to you. If you have any questions **ask**. We are here to help you. 

Here is basic contribution guide: https://github.com/pester/Pester/blob/main/CONTRIBUTING.md

**Comment by efie45** [comment]
Great! I'm stoked to give it a try. Is there a discord or any other channel for communication for questions about contributions or should that just be done here?

**Comment by nohwnd** [comment]
We don't have a special channel just for contributions. You can ask here, or ping me on twitter via DM if you prefer. Or create a discussion in https://github.com/pester/Pester/discussions or ask in the testing channel on slack https://app.slack.com/client/T03MS9QKY/C03QKTUCS

**Comment by fflaten** [comment]
Friendly reminder @efie45. Do you still want to attempt this? 🙂

**Comment by nohwnd** [comment]
Closing this as stale. The original regression that started the thread sorted itself out, and what remained was the idea of a `-ModuleName *` wildcard on `Should -Invoke` to assert calls across any module. I was happy to take that back in 2021 and it was picked up, but it never got implemented and the thread has been quiet since 2022.

I'm going to close it to keep the tracker honest, but the feature idea is still reasonable - if anyone wants to implement a `-ModuleName *` (any-module) wildcard for `Should -Invoke`, a focused PR is very welcome and we can pick the discussion back up there. Thanks!
